mlir/utils/tuna/tuna-script.sh

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
+
+# For installing mysql 8.0 for testing, or for running with an isolated database.
+function mysql_setup_generic
+{
+    # Latest "generic" version.  (Must have libaio1 installed already.)
+    # Note that all this happens without privileges.
+    pushd /tmp
+    wget https://dev.mysql.com/get/Downloads/MySQL-8.0/mysql-8.0.34-linux-glibc2.28-x86_64.tar.gz
+    tar xf mysql-8.0.34-linux-glibc2.28-x86_64.tar.gz
+    mv mysql-8.0.34-linux-glibc2.28-x86_64 mysql
+    export PATH=$PATH:/tmp/mysql/bin
+    mysqld --initialize-insecure --datadir=/tmp/mysql/data
+    # --pid-file=/tmp/mysql/mysqld.pid --socket=/tmp/mysql/mysqld.sock
+    mysqld -D --basedir=/tmp/mysql --datadir=/tmp/mysql/data --log-error=/tmp/mysql/errors.log
+    popd
+
+    # Using this name should force socket access, which we want.
+    TUNA_DB_HOSTNAME=localhost
+    mysql --user root -e 'create database tuna;'
+}
+
+function tuna_setup
+{
+    startdir=`pwd`
+    rm -rf MITuna
+    git clone --branch pf-tuna-rocmlir-3 https://github.com/ROCmSoftwarePlatform/MITuna.git
+    cd MITuna
+    export TUNA_DIR=`pwd`
+
+    python3 -m venv myvenv
+    source myvenv/bin/activate
+
+    # --ignore-installed because of problems upgrading PyYAML.  See also -U.
+    python3 -m pip install -r requirements.txt --ignore-installed
+    python3 -m pip install scipy pandas
+
+    cd $startdir
+
+    export PYTHONPATH=$TUNA_DIR:$PYTHONPATH
+
+    if pgrep mysqld ; then
+        ${TUNA_DIR}/tuna/go_fish.py rocmlir --add_tables
+    fi
+}
+
+function clear_tables
+{
+    tablekind=$1
+
+    if [ "$tablekind" = "convolution" ]; then
+        tablekind="conv"
+    fi
+
+    # config table has foreign keys from job and results tables.  however,
+    # config table doesn't have a session column so we must delete them all.
+    mysql --user root --database tuna -e "delete from rocmlir_${tablekind}_results;"
+    mysql --user root --database tuna -e "delete from rocmlir_${tablekind}_job;"
+    mysql --user root --database tuna -e "delete from rocmlir_${tablekind}_config;"
+}
+
+function tuna_run
+{
+    kind=$1
+    space=$2
+    baselabel=`date --iso-8601=minutes`
```

Comment:
Do the work.  Unfortunately, the tables are differentiated by config-type, so we have separate tables and separate jobs and separate runs for conv, gemm, and attention.  (Maybe we fix that later.)  Label starts with a timestamp to know when these ran.  Configs are global, but other parts are held under the umbrella of a "session" which includes the config-type and tuning-space;  jobs belong to a session and results are looked up by session.  Load-factor is an idea to explore how much parallelism we get and how much non-GPU overhead there is in a tuning run.